src/lib.rs

```diff
@@ -156,6 +156,19 @@ pub enum SanType {
 }
 
 impl SanType {
+	#[cfg(feature = "x509-parser")]
+	fn try_from(name: &x509_parser::extensions::GeneralName<'_>) -> Result<Self, RcgenError> {
```

Comment:
Could you rename this to `try_from_general` or something? `try_from` is very generic, makes mainly sense if part of the TryFrom trait impl. Also: colon placement.